### ISSUE TITLE
Helm v3.0.3 not protecting from overwriting successful deployments

### ISSUE THREAD

**Issue by barry-dow** [helm/helm#7736]
Output of `helm version`:
version.BuildInfo{Version:"v3.0.3", GitCommit:"ac925eb7279f4a6955df663a0128044a8a6b7593", GitTreeState:"clean", GoVersion:"go1.13.6"}
Output of `kubectl version`:
Client Version: version.Info{Major:"1", Minor:"16", GitVersion:"v1.16.1", GitCommit:"d647ddbd755faf07169599a625faf302ffc34458", GitTreeState:"clean", BuildDate:"2019-10-02T17:01:15Z", GoVersion:"go1.12.10", Compiler:"gc", Platform:"darwin/amd64"}
Server Version: version.Info{Major:"1", Minor:"15", GitVersion:"v1.15.5", GitCommit:"20c265fef0741dd71a66480e35bd69f18351daea", GitTreeState:"clean", BuildDate:"2019-10-15T19:07:57Z", GoVersion:"go1.12.10", Compiler:"gc", Platform:"linux/amd64"}

Cloud Provider/Platform (AKS, GKE, Minikube etc.): 
docker-desktop for mac

This is related to https://github.com/helm/helm/issues/3353 which I believe had been fixed in Helm v2.16 (?).

To recreate the problem we used a helm chart that has a bad url for the docker image. To force a successful install we ran without the `--wait` flag:
```
~/helm-v3.0.3/hello-helm>
~/helm-v3.0.3/hello-helm> ../helm install hello-test ./chart/hello
NAME: hello-test
LAST DEPLOYED: Thu Mar  5 15:18:41 2020
NAMESPACE: default
STATUS: deployed
REVISION: 1
TEST SUITE: None
```

We then did several upgrades with `--wait` set, a short timeout and `--history-max` set to 3. For the first two upgrades the successful deployment is still there:
```
~/helm-v3.0.3/hello-helm> ../helm upgrade hello-test ./chart/hello --wait --timeout 1s --history-max 3
Error: UPGRADE FAILED: timed out waiting for the condition
~/helm-v3.0.3/hello-helm> ../helm upgrade hello-test ./chart/hello --wait --timeout 1s --history-max 3
Error: UPGRADE FAILED: timed out waiting for the condition
~/helm-v3.0.3/hello-helm> ../helm history hello-test
REVISION	UPDATED                 	STATUS  	CHART  	APP VERSION	DESCRIPTION
1       	Thu Mar  5 15:18:41 2020	deployed	hello-1	           	Install complete
2       	Thu Mar  5 15:19:20 2020	failed  	hello-1	           	Upgrade "hello-test" failed: timed out waiting for the condition
3       	Thu Mar  5 15:19:24 2020	failed  	hello-1	           	Upgrade "hello-test" failed: timed out waiting for the condition
```

However, this disappears when we do one more upgrade:
```
~/helm-v3.0.3/hello-helm> ../helm upgrade hello-test ./chart/hello --wait --timeout 1s --history-max 3
Error: UPGRADE FAILED: timed out waiting for the condition
~/helm-v3.0.3/hello-helm> ../helm history hello-test
REVISION	UPDATED                 	STATUS	CHART  	APP VERSION	DESCRIPTION
2       	Thu Mar  5 15:19:20 2020	failed	hello-1	           	Upgrade "hello-test" failed: timed out waiting for the condition
3       	Thu Mar  5 15:19:24 2020	failed	hello-1	           	Upgrade "hello-test" failed: timed out waiting for the condition
4       	Thu Mar  5 15:19:52 2020	failed	hello-1	           	Upgrade "hello-test" failed: timed out waiting for the condition
~/helm-v3.0.3/hello-helm>
```

**Comment by jdolitsky** [comment]
@barry-dow - could you see if this is still the case in latest release? (3.1.1)

**Comment by technosophos** [comment]
This was fixed in Helm 3.2.0-rc.1
